pydsa/detecting_cyclicgraph.py

```diff
@@ -0,0 +1,20 @@
+def detecting_cyclicgraph(g):
+	"""Return True if the directed graph g has a cycle.
+	g must be represented as a dictionary mapping vertices to
+	iterables of neighbouring vertices. 
+	"""
+	path = set()
+	visited = set()
+
+	def visit(vertex):
+		if vertex in visited:
+			return False
+		visited.add(vertex)
+		path.add(vertex)
+		for neighbour in g.get(vertex, ()):
```

Comment:
use `None` instead of empty tuple. As it would be easy for others specially beginners to understand the code.